include/eve/arch/riscv/as_register.hpp

```diff
@@ -0,0 +1,497 @@
+    // it requires additional RISC-V extension Zvfh.
+    return;
+  }
+
+  static consteval auto find_vfloat_mf2()
+  {
+    constexpr size_t element_bit_size = sizeof(Type) * 8;
+    if constexpr( element_bit_size == 32 )
+    {
+      using type = vfloat32mf2_t __attribute__((riscv_rvv_vector_bits(__riscv_v_fixed_vlen / 2)));
+      return type {};
+    }
+  }
+
+  static consteval auto find_vfloat_m1()
+  {
+    constexpr size_t element_bit_size = sizeof(Type) * 8;
+    if constexpr( element_bit_size == 32 )
+    {
+      using type = vfloat32m1_t __attribute__((riscv_rvv_vector_bits(__riscv_v_fixed_vlen)));
+      return type {};
+    }
+    else if constexpr( element_bit_size == 64 )
+    {
+      using type = vfloat64m1_t __attribute__((riscv_rvv_vector_bits(__riscv_v_fixed_vlen)));
+      return type {};
+    }
+  }
+
+  static consteval auto find_vfloat_m2()
+  {
+    constexpr size_t element_bit_size = sizeof(Type) * 8;
+    if constexpr( element_bit_size == 32 )
+    {
+      using type = vfloat32m2_t __attribute__((riscv_rvv_vector_bits(__riscv_v_fixed_vlen * 2)));
+      return type {};
+    }
+    else if constexpr( element_bit_size == 64 )
+    {
+      using type = vfloat64m2_t __attribute__((riscv_rvv_vector_bits(__riscv_v_fixed_vlen * 2)));
+      return type {};
+    }
+  }
+
+  static consteval auto find_vfloat_m4()
+  {
+    constexpr size_t element_bit_size = sizeof(Type) * 8;
+    if constexpr( element_bit_size == 32 )
+    {
+      using type = vfloat32m4_t __attribute__((riscv_rvv_vector_bits(__riscv_v_fixed_vlen * 4)));
+      return type {};
+    }
+    else if constexpr( element_bit_size == 64 )
+    {
+      using type = vfloat64m4_t __attribute__((riscv_rvv_vector_bits(__riscv_v_fixed_vlen * 4)));
+      return type {};
+    }
+  }
+
+  static consteval auto find_vfloat_m8()
+  {
+    constexpr size_t element_bit_size = sizeof(Type) * 8;
+    if constexpr( element_bit_size == 32 )
+    {
+      using type = vfloat32m8_t __attribute__((riscv_rvv_vector_bits(__riscv_v_fixed_vlen * 8)));
+      return type {};
+    }
+    else if constexpr( element_bit_size == 64 )
+    {
+      using type = vfloat64m8_t __attribute__((riscv_rvv_vector_bits(__riscv_v_fixed_vlen * 8)));
+      return type {};
+    }
+  }
+
+  static consteval auto find_vfloat()
+  {
+    if constexpr( lmul == 1 ) return find_vfloat_m1();
+    else if constexpr( lmul == 2 ) return find_vfloat_m2();
+    else if constexpr( lmul == 4 ) return find_vfloat_m4();
+    else if constexpr( lmul == 8 ) return find_vfloat_m8();
+    else if constexpr( lmul == -2 ) return find_vfloat_mf2();
+    else if constexpr( lmul == -4 ) return find_vfloat_mf4();
+    // MF8 for float not supported.
+  }
+
+  public:
+  static consteval auto find()
+  {
+    constexpr auto width = sizeof(Type) * Size::value * 8;
+
+    static_assert(width <= __riscv_v_fixed_vlen * ABI::MaxLmul,
+                  "[eve riscv] - Type is not usable in SIMD register (too big)");
```

Comment:
this is fairly internal - I wouldn't worry about it. We have a list of supported types already.

Reply:
I prefer to keep this to be sure that `as_logical_register` is in sync with `as_register`. If you wish, I will delete this.

Reply:
Done